File: 几个随机小数和大于一/几个随机小数和大于一.c
// 计算结果 2.718054 计算耗时 0.490000 秒

#include <stdio.h>
#include <stdlib.h>
#include <time.h>

const int LOOP = 10000000;
int main(void)
{
    srand(clock());
    double start = clock();
    int n = 0;
    for (int i = 0; i < LOOP; ++i)
        for (double x = 0.0; x < 1; x += (double)rand() / RAND_MAX)
            ++n;
    double end = clock();
    printf("计算结果 %.6f 计算耗时 %.6f 秒\n", (double)n / LOOP, (end - start) / CLOCKS_PER_SEC);
}

File: 几个随机小数和大于一/几个随机小数和大于一.py
# 计算结果 2.7178742 计算耗时 5.331498622894287 秒
import random
import time

LOOP: int = 10000000

random.seed(time.time())
start = time.time()
n = 0
for _ in range(LOOP):
    x = 0.0
    while x < 1:
        x += random.uniform(0.0, 1.0)
        n += 1
end = time.time()

print(f"计算结果 {n/LOOP} 计算耗时 {end - start} 秒")
